### PR TITLE
feat: use argparserAlias for xexec

### PR DESCRIPTION
**Pull request by jnoortheen**
## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

File: xonsh/aliases.py [hunk 1]
```diff
@@ -665,44 +678,32 @@ def xexec(args, stdin=None):
...
         return (
             None,
-            "xonsh: exec: file not found: {}: {}"
-            "\n".format(e.args[1], args.command[0]),
+            "xonsh: exec: file not found: {}: {}" "\n".format(e.args[1], command[0]),
             1,
         )
 
 
+xexec = ArgParserAlias(func=xexec_fn, has_args=True, prog="xexec")
```
Comment (daniel-shimon) [inline xonsh/aliases.py:704]: Why not call the function `xexec` and overwrite the variable so that `prog` is deduced?

File: xonsh/cli_utils.py [hunk 2]
```diff
@@ -85,7 +85,21 @@ def _get_param_doc(doc: str, param: str) -> tp.Iterator[str]:
...
+            and set(lin.strip()) == {"-"}
+            and (idx != 0)
+            and (not lines[idx - 1].startswith("Parameters"))
+        ):
+            return os.linesep.join(lines[idx - 1 :])
+
+    return ""
+
+
+def get_doc(func: tp.Union[tp.Callable, str], parameter: str = None, epilog=False):
```
Comment (daniel-shimon) [inline xonsh/cli_utils.py:102]: `epilog` -> `epilogue`

File: xonsh/cli_utils.py [hunk 3]
```diff
@@ -85,7 +85,21 @@ def _get_param_doc(doc: str, param: str) -> tp.Iterator[str]:
         yield lin
 
 
-def get_doc(func: tp.Union[tp.Callable, str], parameter: str = None):
+def _get_epilog_doc(doc: str) -> str:
+    lines = doc.splitlines()
+    for idx, lin in enumerate(lines):
+        if (
+            lin.startswith("-")
```
Comment (daniel-shimon) [inline xonsh/cli_utils.py:92]: `lin.lstrip`?

File: xonsh/cli_utils.py [hunk 4]
```diff
@@ -85,7 +85,21 @@ def _get_param_doc(doc: str, param: str) -> tp.Iterator[str]:
         yield lin
 
 
-def get_doc(func: tp.Union[tp.Callable, str], parameter: str = None):
+def _get_epilog_doc(doc: str) -> str:
+    lines = doc.splitlines()
+    for idx, lin in enumerate(lines):
```
Comment (daniel-shimon) [inline xonsh/cli_utils.py:90]: Isn't there a better way to parse pandoc text?

**Comment by daniel-shimon** [comment]
Why not show the notes section like it appears I nthe documentation, after the parameters?

File: xonsh/aliases.py (hunk 1 above)
Comment (jnoortheen) [reply xonsh/aliases.py:704]: the xexec_fn can be used during tests and other pure Python scripts. Also explicitly we are defining the object.

File: xonsh/cli_utils.py (hunk 2 above)
Comment (jnoortheen) [reply xonsh/cli_utils.py:102]: this is just short

File: xonsh/cli_utils.py (hunk 3 above)
Comment (jnoortheen) [reply xonsh/cli_utils.py:92]: since we use cleandoc, any padding will get removed.

File: xonsh/cli_utils.py (hunk 4 above)
Comment (jnoortheen) [reply xonsh/cli_utils.py:90]: this doesn't need results from other two functions and independent. Let me know if you have any suggestion

**Comment by jnoortheen** [comment]
> Why not show the notes section like it appears I nthe documentation, after the parameters?

After the cli_utils change, it shows the notes section.

File: xonsh/cli_utils.py [hunk 5]
```diff
@@ -105,10 +121,13 @@ def get_doc(func: tp.Union[tp.Callable, str], parameter: str = None):
 
     doc = inspect.getdoc(func) or ""
```
Comment (daniel-shimon) [inline xonsh/cli_utils.py:122]: So call `cleandoc` here before passing the docs to `_get_epilog_doc` and friends

Comment (jnoortheen) [reply xonsh/cli_utils.py:122]: it already does https://docs.python.org/3/library/inspect.html#inspect.getdoc

**Comment by daniel-shimon** [comment]
So why are you running it again letr in the function?

**Comment by jnoortheen** [comment]
> So why are you running it again letr in the function?

the param doc will have some indendation. it is to strip such extra noise.

**Comment by daniel-shimon** [comment]
> the param doc will have some indendation. it is to strip such extra noise.

I understand 
So can you add a comment explaining this?

**Comment by jnoortheen** [comment]
> I understand
> So can you add a comment explaining this?


added

**Review by daniel-shimon — APPROVED** [review]
Thanks!

**Pull request merged**